### PR TITLE
Clementine-playlists to Library integration (lib-redesign branch)

### PR DESCRIPTION
**Pull request by julled**
Added a connection to the clementine player playlists. Is an adaption of the #1738 patch for this branch. 

If the Track of the Clementine Playlist is already in Mixxx trackcollection, the bpm is read from there. That feature is not yet in #1738.

### CODE REVIEW

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you for the work. 

I have just tested it and it works. It is great to see my Clementine playlists im Mixxx  :-)   

The breadcrumb over the track table is not updated. 
There is also a general issue of all library features that a 
"Show all" knot is missing.

File: src/library/features/clementine/clementinefeature.cpp
Comment (julled) [reply src/library/features/clementine/clementinefeature.cpp]: No, i did not miss it, i tried to make a logo, but wasnt able to get the same style. I hope someone who did the other logos will do it...

**Comment by julled** [comment]
Pushed some tabs->spaces patch.
When do you think this will go to the master branch?

**Comment by daschuer** [comment]
After the 2.2 release in September. We can merge this to the library redesign branch whenever it is ready.

**Comment by daschuer** [comment]
Here it is: 
ic_library_clementine

**Comment by julled** [comment]
Cool thx, will add that and do another bugfix...

**Comment by julled** [comment]
@daschuer did you try to use that logo? If i try to use it, strangly it doesnt show up. Other logo files work.

**Comment by julled** [comment]
Nice that was it, thx.

**Comment by julled** [comment]
I think i am done, thx again for your help and your work on mixxx in general!
I am happy i was able to contribute a feature which i always missed...

**Comment by julled** [comment]
fixed

**Comment by daschuer** [comment]
Thank you. We are getting close. Now the breadcrumb works, but it does not show the selected play list like the other external library features. 

Do you have interest to fix the missing "Show all" entry for all external library features? 

After merge, I will put your name into the contributor list into the about box. Is it OK to use your real name there?

**Comment by julled** [comment]
Yes i could do that, but i am confused about the current implementation about that feature for the other players. If you activate at the specific player-connector the "Show All" is activated, but if you activate a specific playlist in that player-connector, there is no way back to "Show All". The only way to get there is to activate another player-connector and then go back. 
Shouldnt be there a specific eye-catching marked entry in the list window for each player-connector for the "show all"?
In the old UI there was the tree-view which made it easy to differentiate between different lists of the connectors.

**Comment by daschuer** [comment]
Yes, the current implementation of the external libraries is somehow halve baked.
I think the best way is to add a topmost artificial "show all" playlist.

I think we can merge this if the playlist name I shown. This way it is head up with the others. 

The "show all" feature can become a new pull request.

**Comment by julled** [comment]
Okey, then i would leave it like it is.

**Comment by julled** [comment]
Hi, is it still possible that this might go to the master branch? Would be sad if not :(

**Comment by Holzhaus** [comment]
@daschuer ping

**Comment by daschuer** [comment]
It is really bad that this has stalled. Unfortunately 2.3 is around the corner and I like to not delay it because of this PR. We can merge it into 2.4.

**Comment by julled** [comment]
Hurray, thank you :)
Will have 2.4 the lib redesigned?
Do you need some additional work? I dont have a mac to get the Travis mac compilation right.

File: src/library/features/clementine/clementinedbconnection.cpp
Comment (Holzhaus) [inline src/library/features/clementine/clementinedbconnection.cpp:142]:
```suggestion
                entry.duration = static_cast<int>(duration / 1000000000);
```

File: src/library/features/clementine/clementinedbconnection.cpp
Comment (Holzhaus) [inline src/library/features/clementine/clementinedbconnection.cpp:132]:
```suggestion
                entry.artist = QStringLiteral("[unknown]");
```

File: src/library/features/clementine/clementinedbconnection.h
Comment (Holzhaus) [inline src/library/features/clementine/clementinedbconnection.h:2]: Use #pragma once instead of oldschool include guards

File: src/library/features/clementine/clementinefeature.cpp
Comment (Holzhaus) [inline src/library/features/clementine/clementinefeature.cpp:65]: Please make this a `const QString kSettingsName` in the anonymous namespace

File: src/library/features/clementine/clementinefeature.h
Comment (Holzhaus) [inline src/library/features/clementine/clementinefeature.h:3]:
```suggestion
#pragma once
```

File: src/library/features/clementine/clementinefeature.h
Comment (Holzhaus) [inline src/library/features/clementine/clementinefeature.h:66]:
```suggestion
```

File: src/library/features/clementine/clementineplaylistmodel.h
Comment (Holzhaus) [inline src/library/features/clementine/clementineplaylistmodel.h:3]:
```suggestion
#pragma once

```

File: src/library/features/clementine/clementineplaylistmodel.h
Comment (Holzhaus) [inline src/library/features/clementine/clementineplaylistmodel.h:49]:
```suggestion
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Please pull latest master into this branch, I hope we can finally get this merged.

**Comment by julled** [comment]
> Please pull latest master into this branch, I hope we can finally get this merged.

thx for the suggestions, i will do these changes in the next weeks

**Comment by uklotzde** [comment]
Unfortunately, this PR has been rebased on the *library redesign* branch according to our request. But that branch is dead. The serious performance issues were unknown back then.

Please rebase your branch again on *master* and then target *master*.

**Comment by julled** [comment]
Okay, i will try that.

**Comment by julled** [comment]
I continued the work on this for a PR to master in #3173 .

**Pull request closed without merging**
